experimental/dds/tree2/docs/undo/v1-undo.md

```diff
@@ -123,52 +127,410 @@ It can be still be thought of as sparse if we ignore the fact that sequenced com
-   then apply the incoming edit(s).
-2. Fork the forest that drives the UI for the parent branch.
+If this switch from not including the refresher with the undo to including it came any later,
+meaning, if we did not include the refresher on undos created beyond the point where the undone edit falls out of the collaboration window,
+then we would run the risk of peers receiving an undo for which they have already discarded the repair data:
 
-Option #2 is bound to be more efficient but may require a more complex contract between parent and child branches.
-These options should be explored during implementation.
+![An undo edit is received by a peer that no longer has the repair data for it](./UndoAfterCollab.png)
+
+In the scenario pictured above, the undo edit must carry with it a refresher for the relevant repair data.
+
+Note that the undo could also end up being sequenced before the point at which peers will garbage-collect the repair data:
+
+![An undo edit is received by a peer that still has the repair data for it](./UndoAfterCollabAndFast.png)
+
+When that's the case, both the peer and the undo edit have some repair data.
+The repair data on both should be the same, so the peer can just ignore the refresher,
+as described in the [Creating Repair Data On Change Application](#creating-repair-data-on-change-application) section.
 
-### Managing Repair Data
+If this switch from not including the refresher along with the undo to including it came any earlier,
+meaning, if we included the refresher on undos created before the point where the undone edit falls out of the collaboration window,
+then we would run the risk of peers receiving an undo with incorrect repair data:
 
-This data can be stored in a repair data store that is co-located with the edit on the relevant commit tree.
-This approach has the following pros and cons:
+![An undo is sent with out of date repair data](./UndoDuringCollab.png)
+
+In the scenario pictured above, the client sending the undo must not include a refresher.
+If it did, and the green edit affected the portion of the document removed by the deletion,
+then the repair data refresher would not reflect the impact of that edit.
+
+It is safe for the undo sent in the scenario pictured above not to include the refresher.
+This is due to the following invariant of the collaboration window:
+If an edit is sent before a given collaboration window span S,
+then it will be sequenced either before or within S, but no later.
+Graphically speaking, this means that if an edit is sent for sequencing within the span of one rectangle,
+then it will be sequenced at the latest within the span of rectangle that follows.
+Here we can apply that rule to see that the undo edit,
+so long as it is conceived within or before the light blue collaboration window span,
+will be sequenced within or before the dark blue collaboration window span,
+during which peers will still be holding on to their copy of the repair data.
```

Comment:
Leaving this comment here so I remember to go back to it once I understand what the dark blue means.